flake.nix

```diff
@@ -532,6 +532,12 @@
               mkdir $out
             '';
 
+        tests.nixpkgsLibTests =
+          nixpkgs.lib.genAttrs systems (system:
+            import (nixpkgs + "/lib/tests/release.nix")
+              { pkgs = nixpkgsFor.${system}; }
+          );
+
```

Comment:
I don't think this test works, because it's evaluated by the Nix of the CI, not the Nix we're building. 

Maybe this should just extend `tests.evalNixpkgs`?

Reply:
> > * `nixpkgsFor` includes our nix by overlay

Just confirmed that it catches the regression. I've also merged a PR upstream so that we can pass it directly, which is more explicit.

> Maybe this should just extend `tests.evalNixpkgs`?

Reusing the whole upstream expression helps with making sure that the release can make its way into Nixpkgs.

Reply:
> the Nix of the CI

The real testing happens in the sandbox. The Nix of the CI only builds the new Nix and provides the environment, but the new Nix is used for the actual testing.